### ISSUE TITLE
publish without Real Molecules screen?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/molecule-polarity#27]
This discussion was relocated from #15.

A 3D viewer is required to complete the Real Molecules screen.  JSmol is not compatible with PhET simulations (see #6), so PhET will need to implement a 3D viewer (see #15).  That will require significant time and effort, and is unlikely to happen for some time. 

In the meantime... Should Molecule Polarity be published without the Real Molecules screen?  

Comments relocated from #15:

@amyh-phet said:
> Mocked up a possibility for a page to put in place for 3rd screen until we get this sorted out. Mockup in Molecule Polarity HTML5 folder on drive. https://drive.google.com/open?id=0BxDjsEf1xP-WY1NYeERLenNGdnM

![screenshot_286](https://user-images.githubusercontent.com/3046552/27608856-366132c4-5b46-11e7-85d6-b2766e7641af.png)

I said:
> I'm confused about the need for this mockup. Has there been a discussion about publishing Molecule Polarity without the "Real Molecules" screen, and with a placeholder like this?

@amyh-phet said:
> @kathy-phet asked me to mock this up as she was considering publishing the sim without the Real Molecules screen. Not sure where this stands, but wanted to make sure the mockup was in an accessible place before I left.

I said:
> The "Real Molecules" screen is already fully implemented, with the exception of the 3D viewer. See screenshot below, or run with `?realMolecules` to see it.  Rather than use a mockup (which isn't translatable), I recommend showing a cleaned up version of what's currently implemented, with a translatable dialog ( "... under development..." ) like the one shown in the mockup. 

![screenshot_287](https://user-images.githubusercontent.com/3046552/27609096-0406f0a6-5b47-11e7-9394-fd76edb7142c.png)

@amyh-phet said:
> @pixelzoom    I will completely defer to you on this one.   I apologize as I wasn't aware of your progress on getting this implemented.  You make an excellent point about this needing to be translatable.

**Comment by pixelzoom** [comment]
In https://github.com/phetsims/molecule-polarity/issues/15#issuecomment-311479894, I said:

The "Real Molecules" screen is already fully implemented, with the exception of the 3D viewer. See screenshot below, or run with `?realMolecules` to see it.  Rather than use a mockup (which isn't translatable), I recommend showing a cleaned up version of what's currently implemented, with a translatable dialog ( "... under development..." ) like the one shown in the mockup. 

![screenshot_287](https://user-images.githubusercontent.com/3046552/27609096-0406f0a6-5b47-11e7-9394-fd76edb7142c.png)

**Comment by pixelzoom** [comment]
In the above commits, I implemented the "under construction" dialog in the Real Molecules screen, shown in the screenshot below.  You can also test drive it in [1.0.0-dev.9](http://www.colorado.edu/physics/phet/dev/html/molecule-polarity/1.0.0-dev.9/molecule-polarity_en.html). 

![screenshot_288](https://user-images.githubusercontent.com/3046552/27647130-ca65747a-5be7-11e7-8171-f8d7ad8b50e7.png)

**Comment by pixelzoom** [comment]
@ariel-phet @emily-phet @kathy-phet Assigning to you to decide whether to move this forward.  If it affects your decision: I have tested this sim extensively, and the code is ready for review.  But implementation of a 3D molecule viewer (#15) is likely a long ways off.

**Comment by emily-phet** [comment]
Publishing without the 3D viewer seems fine to me. It's a bummer, but having the first two screens available is better than nothing. 

- Proposal: I'm not sure it's necessary to include the third screen at all. If there isn't an actual timeline for completion of the third screen, what will we say to people asking when this "Under development" screen will get completed? Unless there's a specific reason to include the screen that provides a benefit to the project (e.g., raising awareness that this third screen requires $$$ to complete), I propose it not be included. It's possible it could be years before that screen is functional (is this correct?). I don't think showing the third screen (without a legitimate timeline for completion) benefits users. The first two screens really do provide a lot of benefit on their own - and it would be better if the sim did not look broken or incomplete.

- If we do include the third screen, it would be nice if the info text was a little friendlier and more specific. For example:
"Real Molecules Screen - Under Development
To receive an update when this screen is complete, sign up for PhET's email list (link here)
Questions? Email phethelp@colorado.edu"

**Comment by ariel-phet** [comment]
I defer to @emily-phet and @kathy-phet. I am usually not a huge fan of publishing a sim and then adding a screen later, but the circumstances of this sim might be an exception.

**Comment by kathy-phet** [comment]
My suggestion would be to include a third screen, but with text that redirects users .... 
something like: 
"The Real Molecules screen is currently not available in HTML5. If your device can run Java, you can access this screen in the legacy Java simulation here: (link to Java)."

**Comment by kathy-phet** [comment]
> @kathy-phet asked me to mock this up as she was considering publishing the sim without the Real Molecules screen. Not sure where this stands, but wanted to make sure the mockup was in an accessible place before I left.

Amy - Sorry about the confusion here. I was trying to suggest mocking up some version of a static screen that would let people know it is not available right now, with some appropriate wording.

**Comment by pixelzoom** [comment]
Questions:

(1) Are we going to proceed with publishing this sim? (I'm not hearing a clear decision.)

If proceeding with publishing:

(2) Who is the  lead designer? I need someone to make decisions.

(3) Shall we omit the Real Molecules screen or display a dialog? (no consensus at the moment).

If showing a dialog:

(4) What is the desired text for the dialog? (no consensus at the moment).

**Comment by emily-phet** [comment]
Regarding @kathy-phet comment above, referring users to Java version sounds fine to me. How about this wording?
"[large text] Real Molecules screen - not available in HTML5
[regular text] The Real Molecules screen is available in the Java version (link).
To check if your device can run Java simulations, see our list of supported devices (link).
Questions? Email phethelp@colorado.edu"

**Comment by pixelzoom** [comment]
@emily-phet said:

> "[large text] Real Molecules screen - not available in HTML5
[regular text] The Real Molecules screen is available in the Java version (link).
To check if your device can run Java simulations, see our list of supported devices (link).

I'm not aware of any common-code support for inserting hyperlinks in scenery Text.  Probably not impossible, but it will add a significant cost. 

> Questions? Email phethelp@colorado.edu"

GitHub is highlighting the email address, so I want to clarify... PhET has no common-code support for creating a link that opens an email client.  Is that something you are asking for, or is it OK if the user needs to copy-paste the email address into their email client?

**Comment by pixelzoom** [comment]
Looks like there's some support for hyperlinks in the About dialog, I'll investigate whether it's been generalized for use in sims.

**Comment by pixelzoom** [comment]
The About dialog uses joist.LinkText.  It lets you associate a link with text, but _entire_ text becomes the hyperlink.  That's not what we want.  And if I break the dialog text into scenery.Text and joist.LinkText chunks, then we can't handle i18n properly.

**Comment by samreid** [comment]
"[large text] Real Molecules screen - not available in HTML5
[regular text] The Real Molecules screen is available in the Java version.
(link)
To check if your device can run Java simulations, see our list of supported devices.
(link)
Questions? We would love to hear from you
(email link)"

**Comment by pixelzoom** [comment]
To clarify... @samreid (in the above comment) is proposing to put hyperlinks on separate lines.   This will allow us to use joist.LinkText for the hyperlinks.   It requires 4 separate strings for translation, one for each line. 

Notes about an email link...
• It could be added using joist.LinkText by doing something like `new LinkText( 'phethelp@colorado.edu', 'mailto:phethelp@colorado.edu', {...} )`.
• Clicking on this link will open the local email client.  Does that work on all supported platforms? What does it do on Chromebooks? (This will require testing on all platforms.)

**Comment by pixelzoom** [comment]
I'm not getting timely feedback or a clear message on what to do here.

@ariel-phet please pick one of the following, so that I know how to proceed with this issue.

- [ ] Publishing with the first 2 screens.
- [ ] Publish after we complete the "Real Molecules" screen.
- [ ] No decision; defer and revisit on ________ (date).

**Comment by kathy-phet** [comment]
I picked publish with the first 2 screens above.

Emily has a paper due today, and we haven't had time to discuss, sorry. We will have a phone call, and try to fit it in at the end of that.

**Comment by kathy-phet** [comment]
> To clarify... @samreid (in the above comment) is proposing to put hyperlinks on separate lines. This will allow us to use joist.LinkText for the hyperlinks. It requires 4 separate strings for translation, one for each line.
> 

It what Sam describes possible then? Providing a way to add hyperlinks.

**Comment by pixelzoom** [comment]
@kathy-phet asked:
> Is what Sam describes possible then? Providing a way to add hyperlinks.

Yes - if we put the links on separate lines, so that we use existing functionality in joist.  And it's unclear when the email link will work on all supported platforms.

**Comment by ariel-phet** [comment]
@kathy-phet @pixelzoom @samreid @emily-phet personally I would nix the email link. Seems complicated and unnecessary

**Comment by pixelzoom** [comment]
For these proposed lines:
> To check if your device can run Java simulations, see our list of supported devices.
> (link)

There is no link that provides "our list of supported devices".  The closest I can come is either:

(1)  the System Requirements FAQ at https://phet.colorado.edu/en/help-center/running-sims/general#q11-header

(2) the legacy sim's System Requirements at https://phet.colorado.edu/en/simulation/legacy/molecule-polarity#requirements-header

According to @mattpen, neither of these links will reportedly work on IE or Edge. (Links with collapsible divs, like `#requirements-header` apparently don't work on Microsoft browsers.)

And we should reword the text to:
> To check if your device can run Java simulations, see the System Requirements:

**Comment by pixelzoom** [comment]
Here's a dev version that provides a link to the Java version and an email link.   The "supported devices" was a can of worms, so I deleted it. 

Dev version:
http://www.colorado.edu/physics/phet/dev/html/molecule-polarity/1.0.0-dev.10/molecule-polarity_en.html 

Screenshot:
![screenshot_292](https://user-images.githubusercontent.com/3046552/27878699-575a189c-617c-11e7-8ff9-5df8d2f91d68.png)

@phet-steele If you have a minute, could you test the email link (phethelp@colorado.edu) on a Chromebook and describe what happens?  If it's significant, test both when logged into Google, and when not logged in.

**Comment by phet-steele** [comment]
> could you test the email link (phethelp@colorado.edu) on a Chromebook and describe what happens?



> when logged into Google

Starts an email draft (in a new tab in the browser)



> when not logged in

Asks you to sign in/create a google account (in a new tab). Signing in does not direct to an email draft. The link in the sim must be clicked again.

**Comment by kathy-phet** [comment]
Review by Emily/Kathy: 
- Remove "Questions?" line and email link. 
- Change first line to "Real Molecules screen - Not currently available in HTML5"
thanks!

**Comment by pixelzoom** [comment]
Here's another dev version with these changes from 7/6/17 status meeting:

• Removed "Questions?" text and email link (due to issues on Chromebooks and possibly other platforms)
• Changed first line to read "Real Molecules is not currently available in HTML5."
• Changed second line to read "If your device runs Java, you can use the Java version at"
• Remove all UI elements that were visible behind the dialog

http://www.colorado.edu/physics/phet/dev/html/molecule-polarity/1.0.0-dev.11/molecule-polarity_en.html

![screenshot_293](https://user-images.githubusercontent.com/3046552/27924254-db508f7a-623d-11e7-9dc7-af8921036ba0.png)

**Comment by kathy-phet** [comment]
This looks great to me. Closing.
